sopel/modules/url.py

```diff
@@ -360,6 +367,11 @@ def process_urls(
         # FIXME: This does nothing when an attacker knows how to host a 302
         # FIXME: This whole concept has a TOCTOU issue
         if not bot.config.url.enable_private_resolution:
+            if not parsed_url.hostname:
```

Comment:
I'm guessing this is an unrelated bugfix?

Reply:
Nope, `hostname` is `Optional[str]`, because `file:///path` will have `None` for the hostname, and yes this is a valid URL. However, `ip` requires `str`, so mypy legitimately complains about it.